homeassistant/components/watergate/__init__.py

```diff
@@ -0,0 +1,101 @@
+"""The Watergate integration."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from http import HTTPStatus
+import logging
+
+from watergate_local_api import WatergateLocalApiClient
+from watergate_local_api.models import WebhookEvent
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.webhook import Request, Response, async_generate_url
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME, CONF_WEBHOOK_ID, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import WatergateDataCoordinator
+from .entity import WatergateConfigEntry, WatergateData
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS: list[Platform] = [
+    Platform.VALVE,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: WatergateConfigEntry) -> bool:
+    """Set up Watergate from a config entry."""
+    sonic_name = entry.data[CONF_NAME]
+    sonic_address = entry.data[CONF_IP_ADDRESS]
+    webhook_id = entry.data[CONF_WEBHOOK_ID]
+
+    _LOGGER.debug(
+        "Setting up watergate local api integration for device: %s (IP: %s)",
+        sonic_name,
+        sonic_address,
+    )
+
+    watergate_client = WatergateLocalApiClient(sonic_address)
+
+    coordinator = WatergateDataCoordinator(hass, watergate_client)
+
+    hass.components.webhook.async_unregister(webhook_id)
+    hass.components.webhook.async_register(
+        DOMAIN, "Watergate", webhook_id, get_webhook_handler(coordinator)
+    )
+
+    _LOGGER.debug("Registered webhook: %s", webhook_id)
+
+    await coordinator.async_config_entry_first_refresh()
+
+    await watergate_client.async_set_webhook_url(
+        async_generate_url(hass, webhook_id, allow_ip=True)
+    )
+
+    entry.runtime_data = WatergateData(watergate_client, coordinator, sonic_name)
```

Comment:
You can reach the client via the `coordinator.api`, should we remove the dataclass wrapper and only put the coordinator in here?